### ISSUE TITLE
`less -FRX` drops keypresses when not paging due to `-F`

### ISSUE THREAD

**Issue by divVerent** [gwsw/less#599]
In the past, when `less -FRX` was used (such as default in `git`'s and `hg`'s pager invocations), if the output was short, and no paging was necessary, keys pressed during the command invocation ended up in the shell prompt.

Now, `less -FRX` drops a single keypress typed during this time.

Repro:

```
$ { echo x; sleep 1; echo y; } | ./less -FRX
```

Then hit enter, and immediately type `1234` before the next prompt appears. The prompt will show just `234`.

I bisected the issue to

```
commit b17b0802471c8b30d4f2f5085f621a5ea9e2d052
Author: Mark Nudelman <markn@greenwoodsoftware.com>
Date:   Wed Nov 16 15:19:28 2022 -0800

    Fix problems with EOF after new poll behavior
    in 953257f9030d29dd930ed3e76fd5811681b672b1.
```

but also uncovered other broken behaviors along the way in history:

- The version just before this commit doesn't eat keypresses, but it also leaves out the last line (the `y`) in my example from the output. That's the bug this commit fixed, I suppose.
- Some newer versions actually eat _all_ keypresses while `less` runs, and shows a prompt that one can cancel using `^X`. In a way that's actually better, as it at least shows some UI making one "expect" this behavior.
- In current `master`, I get yet another different behavior: `less` now eats all keypresses too, and starts its `:` prompt - which means that if any key was pressed while `less` runs, the command won't exit at all until one exits that command line again.

Can we somehow get the old behavior restored, as it fits much better into tools like `git` and `hg` where the user often knows in advance whether the output will be more or less than one screen, and can continue working accordingly without waiting for the pager, whereas these tools like to just send ALL their output to the pager unconditionally, as they can't know in advance whether it'll be short or long?

If not, I'll probably have to write my own wrapper that first reads the first screenful, prints it to the screen, and once more data arrives, actually invokes less on this data + anything more coming in. Seems overly complex to have to do that, as the whole point of `-F` aka `--quit-if-one-screen` is to do just that...

**Comment by gwsw** [comment]
I can't reproduce this from your description, but it sounds like the same issue as #557. Is it resolved if you use the --no-poll option?

**Comment by divVerent** [comment]
Confirmed that `--no-poll` restores the good behavior. Sadly it's not available on the systems I need it, but it's not like it's hard to install my own build of `less`.

**Comment by divVerent** [comment]
Confirmed

```
[pager]
pager = LESSKEY_CONTENT="$LESSKEY_CONTENT;#env;#version>=668 LESS=\$\{LESS\} --no-poll" less -FRX
```

working, and it _should_ also preserve existing use of LESSKEY_CONTENT. Not sure if using that variable breaks existing users of the LESSKEY or LESSKEYIN variables.

**Comment by divVerent** [comment]
`git` seems to need:

```
[core]
        pager = "LESSKEY_CONTENT=\"$LESSKEY_CONTENT;#env;#version>=668 LESS=\\${LESS} --no-poll\" less"
```

**Comment by gwsw** [comment]
Seems like this is resolved. Any changes to git or other programs are up to the developers of those programs of course, although I would hope they will make any such changes optional, for those of us who do not want the --no-poll behavior.

**Comment by divVerent** [comment]
Resolved by a change that restores the previous default, or resolved by doing nothing and expecting third party tools to add special logic for compatibility to newer `less`?

**Comment by gwsw** [comment]
I had intended the latter (the user would use --no-poll when they want this behavior), but after reconsideration I have made a change (52b2d920f492ee93a23996cf9a4d931ad0e96a21) that disables polling when -F is in effect and the file is short. I think this eliminates the need for the --no-poll option and I plan to remove it soon.

**Comment by divVerent** [comment]
Thank you! Much appreciated.

As for the option, if some people really want ^X to be understood during the initial phase of -F, maybe the polling option should be a tristate (i.e. -F changes the default, but --poll and --no-poll override it)?

Personally though, I do not think many people will want both `-F` and `^X` support at once, as the whole point of `-F` is to not have any user interface if the input is short (e.g. tailing a slowly growing file into `less` works just fine without `-F`).

**Comment by gwsw** [comment]
Yes, I agree that it's unlikely that a user would want to input ^X or other keys while -F is in effect for a short file. When displaying such a file, less will exit immediately after printing the file contents, so there would be only a small fraction-of-a-second window in which the user could enter anything. That's why I think that --no-poll is no longer needed.
